pyo3-ffi/src/methodobject.rs

```diff
@@ -71,11 +78,44 @@ extern "C" {
 #[derive(Copy, Clone)]
 pub struct PyMethodDef {
     pub ml_name: *const c_char,
-    pub ml_meth: Option<PyCFunction>,
+    pub ml_meth: MlMeth,
     pub ml_flags: c_int,
     pub ml_doc: *const c_char,
 }
 
+/// Function types used to implement Python callables.
+///
+/// This union must be accompanied by the correct [ml_flags](PyMethodDef::ml_flags),
+/// otherwise the behavior is undefined.
+///
+/// See the [Python C API documentation][1] for more information.
+///
+/// [1]: https://docs.python.org/3/c-api/structures.html#implementing-functions-and-methods
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub union MlMeth {
```

Comment:
I don't particularly like the name. Can we call it `PyMethodDefPointer` (similar to a union we already have without equivalent in CPython, `PyUnicodeObjectData`)?